File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystAnnouncement.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable
import kotlin.time.ExperimentalTime

/**
 * A platform-wide announcement
 */
@Serializable
data class CatalystAnnouncement @OptIn(ExperimentalTime::class) constructor(
    val id: String,
    val title: String,
    val body: String,
    val category: String,
    val since: kotlin.time.Instant,
    val until: kotlin.time.Instant,
    val url: String? = null
)

/**
 * Wrapper for the `{"announcements": [...]}` response shape
 */
@Serializable
internal data class CatalystAnnouncementsWrapper(val announcements: List<CatalystAnnouncement>)

File: packages/swift/Sources/CatalystSwift/Models/CatalystAnnouncement.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

import Foundation

/// A platform-wide announcement.
public struct CatalystAnnouncement: Decodable, Sendable, Equatable, Hashable {
  public let id: String
  public let title: String
  public let body: String
  public let category: String
  public let since: Date
  public let until: Date
  public let url: String?
}

public struct CatalystAnnouncementsWrapper: Decodable, Sendable {
  public let announcements: [CatalystAnnouncement]
}
